doc/man3/flux_requeue.rst

```diff
@@ -10,29 +10,31 @@ SYNOPSIS
 
   #include <flux/core.h>
 
-  int flux_requeue (flux_t *h, const flux_msg_t *msg, int flags);
+  cc $(pkg-config --cflags --libs flux-core)
```

Comment:
[man(7)](https://man7.org/linux/man-pages/man7/man-pages.7.html) says for the SYNOPSIS section:

>  For functions, it shows any required data
>  declarations or #include directives, followed by the
>  function declaration.

I'm not sure a plain compile line covers that. But didn't we have some issue with sphinx/reStructuredText that fails to render multiple lines here? Maybe I'm getting that confused with something else.

Anyway, I'm not sure if replacing the function prototype with a compile line is the right direction? (Sorry, I probably just don't know the impetus for this change.)

Reply:
I was thinking of how to document some classes like idset or flux_msg_t that may have dozens of functions.  At times it is nice to have man pages broken up into one page per one or a few functions.  At other times it is nice to have the whole class in one place.  (This particular case is not a good example of that problem).

Maybe it would be better to do "class docs" as separate RTD pages and then just pick the most commonly used to make into man pages?

Reply:
Oh I should check if a long function gets truncated in the man page rendering.  It apparently wraps in HTML (see PMI RFC - the Spawn call in there is ridiculously long).  Anyway maybe this is not the way to go for man pages.

Reply:
As an example, the `printf(3)` family of functions has a SYNOPSIS like so:
```
SYNOPSIS
       #include <stdio.h>

       int printf(const char *format, ...);
       int fprintf(FILE *stream, const char *format, ...);
       int dprintf(int fd, const char *format, ...);
       int sprintf(char *str, const char *format, ...);
       int snprintf(char *str, size_t size, const char *format, ...);

       #include <stdarg.h>

       int vprintf(const char *format, va_list ap);
       int vfprintf(FILE *stream, const char *format, va_list ap);
       int vdprintf(int fd, const char *format, va_list ap);
       int vsprintf(char *str, const char *format, va_list ap);
       int vsnprintf(char *str, size_t size, const char *format, va_list ap);
```

Does this not cover the whole class in one page case? (Sorry If I'm just being slow)